File: academic-system-python/src/academic_system/security/session.py
from __future__ import annotations
from typing import Optional
from academic_system.model.user import User
from academic_system.model.role import Role
import logging

logger = logging.getLogger(__name__)


class Session:
    _current_user: Optional[User] = None

    @classmethod
    def login(cls, user: User):
        cls._current_user = user

    @classmethod
    def logout(cls):
        if cls._current_user:
            logger.info(f"Logout: username={cls._current_user.username}, "
                        f"role={cls._current_user.role}")
        cls._current_user = None

    @classmethod
    def get_current_user(cls) -> Optional[User]:
        return cls._current_user

    @classmethod
    def get_current_role(cls) -> Optional[Role]:
        return cls._current_user.role if cls._current_user else None

    @classmethod
    def is_logged(cls) -> bool:
        return cls._current_user is not None

    @classmethod
    def is_admin(cls) -> bool:
        return (cls._current_user is not None and
                cls._current_user.role == Role.ADMIN)

File: academic-system/src/main/java/org/example/academic/system/security/Session.java
package org.example.academic.system.security;

import org.example.academic.system.model.Role;
import org.example.academic.system.model.User;

public class Session {

    private static User currentUser;

    public static void login(User user) {
        currentUser = user;
    }

    public static void logout() {
        currentUser = null;
    }

    public static User getCurrentUser() {
        return currentUser;
    }

    public static Role getCurrentRole() {
        return currentUser != null ? currentUser.getRole() : null;
    }

    public static boolean isLogged() {
        return currentUser != null;
    }

    public static boolean isAdmin() {
        return currentUser != null && currentUser.getRole() == Role.ADMIN;
    }
}